src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -0,0 +1,382 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Featurizer.h"
+#include "../Archive.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+//the following functions are introduced from RedSVD
+//Copyright attached
+/*
+ * A header-only version of RedSVD
+ *
+ * Copyright (c) 2014 Nicolas Tessore
+ *
+ * based on RedSVD
+ *
+ * Copyright (c) 2010 Daisuke Okanohara
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above Copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above Copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the authors nor the names of its contributors
+ *    may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ */
+template<typename Scalar>
+inline void sample_gaussian(Scalar& x, Scalar& y)
```

Comment:
>sample_gaussian [](start = 12, length = 15)

It looks like these functions are only used in the implementation. If that is the case, they can be moved until after the declaration of the different classes but before their declarations.